R/zchunk_L2322.Fert_CHINA.R

```diff
@@ -0,0 +1,499 @@
+
+    # Select the supply sector information for fertilizer sector within the US and expand to all of the
+    # sates that are fertilizer producers, then create subsector from province and fertilizer name.
+    L2322.Supplysector_Fert %>%
+      filter(region == gcamchina.REGION, supplysector == "N fertilizer") %>%
+      select(region, supplysector) %>%
+      repeat_add_columns(Fert_provinces) %>%
+      mutate(subsector = paste(province, "N fertilizer")) ->
+      L2322.Supplysector_Fert_provinces
+
+    # Now add the logit table information to the province fertilizer supply sector data frame.
+    L2322.Supplysector_Fert_provinces %>%
+      mutate(logit.year.fillout = min(MODEL_YEARS),
+             logit.exponent = gcamchina.FERT_LOGIT_EXP,
+             logit.type = NA) %>%
+      select(region, supplysector, subsector, logit.year.fillout, logit.exponent, logit.type) ->
+      L2322.SubsectorLogit_CHINAFert
+
+    # Create the subsector default share-weights for the using the min base years
+    # for the fill out year.
+    L2322.SubsectorLogit_CHINAFert %>%
+      select(region, supplysector, subsector) %>%
+      mutate(year.fillout = min(MODEL_BASE_YEARS),
+             share.weight = 1) ->
+      L2322.SubsectorShrwtFllt_CHINAFert
+
+    # Use the subsector logit exponents of fertilizer sector to create
+    # a table of the subsector default technology share-weights for the US.
+    # that will be interpolated.
+    L2322.SubsectorLogit_CHINAFert %>%
+      select(region, supplysector, subsector) %>%
+      mutate(apply.to = "share-weight",
+             from.year = max(MODEL_BASE_YEARS),
+             to.year = max(MODEL_YEARS),
+             interpolation.function = "fixed") ->
+      L2322.SubsectorInterp_CHINAFert
+
+
+    # Expand the supply sector and subsector share weights to all model years.
+    L2322.SubsectorLogit_CHINAFert %>%
+      select(region, supplysector, subsector) %>%
+      mutate(technology = subsector) %>%
+      repeat_add_columns(tibble(year = MODEL_YEARS)) %>%
+      mutate(share.weight = 1) ->
+      L2322.TechShrwt_CHINAFert
+
+
+    # Subset the province fertilizer production data for model base years,
+    # format digits, and add region and supplysector information to prepare
+    # the data frame to add logit table information.
+    L1322.out_Mt_province_Fert_Yh %>%
+      filter(year %in% MODEL_BASE_YEARS) %>%
+      mutate(calOutputValue = signif(value, aglu.DIGITS_LAND_USE)) %>%
+      select(-value) %>%
+      mutate(region = gcamchina.REGION, supplysector = "N fertilizer") %>%
+      unite(subsector, province, supplysector, sep = " ", remove = FALSE) ->
+      L2322.Production_CHINAFert
+
+    # Add technology and subsector share weights and other logit table
+    # information to the calibrated output production for fertilizer in the
+    # CHINA region fertilizer sector.
+    L2322.Production_CHINAFert %>%
+      mutate(technology = subsector,
+             input = "N fertilizer",
```

Comment:
may want to define "N fertilizer" as a constant because it gets used multiple times